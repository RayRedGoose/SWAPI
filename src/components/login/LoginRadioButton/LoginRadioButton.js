import React from 'react';
import PropTypes from 'prop-types'
import './LoginRadioButton.scss'

const LoginRadioButton = ({type, currentRank, toggleChangeState}) => {
  const handleChanges = (event) => {
    toggleChangeState(type, event.target.value)
  }

  return (
    <div className="radio-button-wrapper">
      <input
        id={type}
        type="radio"
        value={type}
        onChange={handleChanges}
        checked={currentRank === type} />
      <label htmlFor={type}>{type.charAt(0).toUpperCase() + type.slice(1)}</label>
    </div>
  )
}

export default LoginRadioButton

LoginRadioButton.propTypes = {
  type: PropTypes.string,
  currentRank: PropTypes.string,
  toggleChangeState: PropTypes.func
}
