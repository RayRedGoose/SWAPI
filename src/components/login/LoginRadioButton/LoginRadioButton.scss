$font-dark: #5A2C05;

.radio-button-wrapper {
  input {
    position: absolute;
    visibility: hidden;
  }
  label {
    font-size: 18px;
    color: $font-dark;
    opacity: 0.5;
    padding: 0.4em 1em;
  }
  input:checked + label {
    border: 2px solid $font-dark;
    opacity: 1;
  }
}
