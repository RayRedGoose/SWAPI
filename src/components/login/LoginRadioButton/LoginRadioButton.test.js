import React from 'react'
import {shallow} from 'enzyme'
import LoginRadioButton from './LoginRadioButton'

describe("LoginRadioButton", () => {
  let mockProps, radio
  const toggleChangeState = jest.fn()

  beforeEach(() => {
    mockProps = {
      key: 'expert',
      type: 'expert',
      currentRank: 'novice',
      toggleChangeState: toggleChangeState
    }
    radio = shallow(<LoginRadioButton {...mockProps} />)
  })

  it ('should match the snapshot with all data passed in correctly', () => {
    expect(radio).toMatchSnapshot()
  })

  it('should call toggleChangeState prop when radio button is changed', () => {
    radio.find('input').simulate(
      'change',
      { target: { value: 'novice'}
    })

    expect(toggleChangeState).toHaveBeenCalled()
  })
})
