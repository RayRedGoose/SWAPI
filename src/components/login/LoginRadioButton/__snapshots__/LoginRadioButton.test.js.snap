// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginRadioButton should match the snapshot with all data passed in correctly 1`] = `
<div
  className="radio-button-wrapper"
>
  <input
    checked={false}
    id="expert"
    onChange={[Function]}
    type="radio"
    value="expert"
  />
  <label
    htmlFor="expert"
  >
    Expert
  </label>
</div>
`;
